packages/flutter/test/material/tabs_test.dart

```diff
@@ -4851,7 +4852,10 @@ void main() {
     );
 
     expect(tester.widget<InkWell>(find.byType(InkWell)).splashFactory, splashFactory);
-    expect(tester.widget<InkWell>(find.byType(InkWell)).overlayColor, overlayColor);
+    expect(
+      tester.widget<InkWell>(find.byType(InkWell)).overlayColor.runtimeType.toString(),
```

Comment:
Why can't this just checked against the local `overlayColor` like before? Why the check for the data type by string?

Reply:
This is because I'm using `_EffectiveOverlay` class to provide `selected` states to the `overlay` property
https://github.com/flutter/flutter/pull/116110/files#diff-b90db0bc5b533dbd8ef4c5099057b292200e0aa96125e74ab3646c783b468bbdR1383-R1389


When doing so, existing tests fail with this message
```console
Expected: <Instance of '_MaterialStatePropertyWith<Color?>'>
  Actual: <Instance of '_EffectiveOverlay'>
```

As a result, I am checking the `runtimeType` instead and which confirms `InkWell` does have an overlay. 
is there any to add `MaterialState.selected` to the existing `Inkwell.overlay` without creating a custom?  I have tired statesController but it removes other states.

Reply:
`ToggleButtons` also does this, it has a custom class for the selected state.

Reply:
I see. I was able to use the states controller in `SegmentedButton`:

https://github.com/flutter/flutter/blob/853b3080e096abb0479f2b5023b4a389e65a27ae/packages/flutter/lib/src/material/segmented_button.dart#L319-L323

But perhaps that won't work with the tab bar entries.

Reply:
Thanks, I've looked into this.
`SegmentedButton` provides the selected color to a Material widget when it has `MaterialState.selected`.

This is the flow:
`SegmentedButton` (this sets the state) -> `TextButton`-> `ButtonStyleButton` (This gets the states and applies 
background color to the `Material.color` widget. 

https://github.com/flutter/flutter/blob/17c1dbc47378da46bb4472dd77a6b12158fec48e/packages/flutter/lib/src/material/button_style_button.dart#L382
 
`Tabs` doesn't have its own `Material`, it just has `InkWell`, it doesn't have background color property so I'm updating `InkWell.overlayColor`  to have `selected` on top of existing `hovered`, `focus`, `pressed` states for selected tab. 

I just refactored overlay code to provide selected state for default overlay and removed `_EffectiveOverlay` class so this test no longer need to updated since this test overlay from the theme.